SparkPost/2web2ui
# SA-1759: Add Subject Campaign Filters to D12Y Metrics

### What Changed

- Changed the filter type value in context to always be filter type key rather than label.
- Added subject campaign filter/group by
- Added conditions to show subject campaign.

### How To Test

- Enable d12y product.
- Go to signals/analytics
- On the default metrics, verify that subject campaign is not an option in filters and group by.
- Remove all of the metrics and only add a d12y metric.
- Check that subject campaign is available in filters and group by
- Add a subject campaign as a filter and verify it works.
- Remove it and add a subject campaign as a comparison. Verify it works.
- Verify that you cannot add any more non d12y metrics.
- Remove the subject campaign comparison. Add a non d12y metric.
- Verify that subject campaign is no longer available as a filter or group by option.
 
### To Do

  > Manage reviewers' expectations on missing features, tests, bugs, late design changes, ...)
